Doc: add a page about security issues (#10844)
OSGeo/gdal: doc/source/user/security.rst

  method.

- Drivers depending on third-party libraries whose code has been embedded in GDAL.
  Binary builds might rely on the internal version, or the external version.
  If using the internal version, they might use an obsolete version of the
  third-party library that might contain known vulnerabilities. Potentially
  concerned drivers are GTiff (libtiff, libgeotiff), PNG (libpng), GIF (giflib),
  JPEG (libjpeg), PCRaster (libcsf), GeoJSON (libjson-c), MRF (liblerc).
  An internal version of ZLib is also contained in GDAL sources.
  Packagers of GDAL are recommended to use the external version of the libraries
  when possible, so that security upgrades of those dependencies benefit to GDAL.

- Drivers using GDALOpen()/GDALOpenEx()/OGROpen() internally cause other drivers
  to be used (and their possible flows exploited), without it being obvious at
  first sight. VRT, MBTiles, KMLSuperOverlay, RasterLite, PCIDSK, PDF, RPFTOC,
  RS2, WMS, WCS, WFS, OAPIF, OGCAPI, GTI ... are such drivers.

- Drivers depending on downloaded data (HTTP, WMS, WCS, WFS, OAPIF, OGCAPI,
  STACIT, STACTA, etc.).

- XML based drivers: they might be subject to denial of service by

>   :ref:`​billion laugh-like <https://en.wikipedia.org/wiki/Billion_laughs_attack>`__

Reviewer: ```suggestion
  `​billion laugh-like <https://en.wikipedia.org/wiki/Billion_laughs_attack>`__
```

.. _security:

================================================================================
Security considerations
================================================================================

This page discusses some security issues that users or developers may face while
using GDAL. It can also serve as a check-list of items to take care of, for
developers coding new drivers.

While GDAL maintainers take a number of steps to limit the risk of software
vulnerabilities, GDAL has a large attack surface, because it deals with data in
a variety of file formats.

.. note::

>     This page is not meant as being a comprehensive cook-book for being "safe" in all

Reviewer: ```suggestion
    This page is not intended to be a comprehensive cook-book for being "safe" in all
```

  size might be suspicious (but not always, for example a VRT file, or highly
  compressed data...)

- Do not allow arbitrary arguments to be passed to command line utilities.
  In particular ``--config GDAL_DRIVER_PATH xxx`` or ``--config OGR_DRIVER_PATH xxx``
  could be used to trigger arbitrary code execution for someone with write
  access to some parts of the file system. Similarly, :config:`GDAL_VRT_ENABLE_PYTHON`
  or :config:`GDAL_VRT_PYTHON_TRUSTED_MODULES` could be used to trigger hostile
  Python code execution.

- When possible, avoid using closed-source dependency libraries that cannot be
  audited for vulnerabilities.

.. _security_known_issues:

Credential related issues
-------------------------

GDAL network virtual file systems can access credentials needed to grant access
to remote resources, from the environment or local files. Those secrets are
stored in RAM using normal mechanisms. Consequently they could potentially be
recovered by untrusted code that would run in the same process as GDAL code.

The same applies for connection strings to some drivers such as PostgreSQL, MySQL, etc.
which my include passwords.

> 

Reviewer 2: my ->may

Author: fixed, thanks
